mesonbuild/scripts/python_info.py

```diff
@@ -63,6 +63,34 @@ def links_against_libpython():
     from distutils.sysconfig import get_config_var
     suffix = get_config_var('EXT_SUFFIX')
 else:
+    # Determine some information from the sys_root if it is given as the first
+    # argument. When not given, it is expected to be an empty string.
+    if len(sys.argv) >= 1 and sys.argv[1]:
+        import importlib.util, pathlib
+        try:
+            # We need to load the sys_root sysconfigdata from its stdlib
+            # directory. We assume that the same Python version as the build
+            # host exists within the sys_root with the same stdlib directory,
+            # save for the prefix. If not, we fall back to the build host.
+            stdlib = sysconfig.get_path('stdlib', vars={'installed_base': sys.argv[1]})
+            # We cannot predict exactly what the sys_root sysconfigdata filename
+            # will be, but there should only be one, so glob to find it.
+            sysconfigdatas = list(pathlib.Path(stdlib).glob('_sysconfigdata_*.py'))
+
+            # Only use the sys_root sysconfigdata if we find exactly one. More
+            # than one is odd, so it is safer to fall back to the build host.
+            if len(sysconfigdatas) == 1:
+                data_spec = importlib.util.spec_from_file_location('_sysconfig_data', sysconfigdatas[0])
+                data_mod = importlib.util.module_from_spec(data_spec)
+                data_spec.loader.exec_module(data_mod)
```

Comment:
I personally don't care about anything older than 3.8, and certainly not 2.7, so I thought I'd save myself any trouble there, but this is a fair question. I took this approach from gpep517. @mgorny, is there a particular reason you did it this way? I'm guessing you didn't care about anything that old either.

Reply:
I think I've just searched for a way to import a module by path since, well, that's precisely what we wanted to do. We don't support older Python versions, so I didn't give it a second thought.

Reply:
I've reworked this to use `exec` and tested against Python 2.7 as well as 3.11. I was initially concerned that it wouldn't be able to handle the Windows <3.8.7 case while using a sysroot, but it turns out Windows doesn't have sysconfigdata files anyway.